jujugui/static/gui/src/app/components/machine-view/machine-view.js

```diff
@@ -35,7 +35,43 @@ YUI.add('machine-view', function() {
       @returns {String} The intial state.
     */
     getInitialState: function() {
-      return {selectedMachine: null};
+      return {selectedMachine: this._getFirstMachineId(this.props.machines)};
+    },
+
+    /**
+      Called when the component is supplied with new props.
+
+      @method componentWillReceiveProps
+      @param {Object} The new props.
+    */
+    componentWillReceiveProps: function(nextProps) {
+      var selectedMachine = this.state.selectedMachine;
+      if (selectedMachine) {
+        // Check that the machine still exists in the db (in case the machine
+        // has been deployed and has been given a new id).
```

Comment:
The id for the currently selected machine is stored as selectedMachine e.g. new5. If the machine is deployed then there will be no machine with the id of new5 (it will have an id of 5 or some other random number) so we we need to check if the machine exists and if it doesn't then select the first machine.